.github/workflows/generate.yml

```
@@ -0,0 +1,40 @@
+#
+# Copyright 2022 The Sigstore Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Generate code
+
+permissions:
+  contents: read
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
```

Assuming we want this running on all pull requests, prior to merge, I think this needs to not have a `branches` key.
```suggestion
  pull_request: {}
```

---

I'm not an Actions expert, so you might need to play with these settings a bit until it starts building this PR.

---

Ohhh i think i can leave it blank maybe, with the paths

---

I think the branches is for targetted against, so understood if it's meant to capture all branches.

Either new workflows aren't triggered in the pull request itself or I've done something wrong still

---

Yeah, maybe it needs to be merged before it starts working. We can fix in a follow-up if not!